core/domain/stats_domain.py

```diff
@@ -403,6 +405,155 @@ def validate(self):
                     '%s cannot have negative values' % (stat_property))
 
 
+class ExplorationIssue(object):
+    """Domain object representing an exploration issue."""
+
+    def __init__(self, issue_id, schema_version, customization_args):
+        """Constructs an ExplorationIssue domain object.
+
+        Args:
+            issue_id: str. ID of the Issue.
+            schema_version: int. Schema version of the customization args dict.
+            customization_args: dict. The customization dict. The keys are
+                names of customization_args and the values are dicts with a
+                single key, 'value', whose corresponding value is the value of
+                the customization arg.
+        """
+        self.issue_id = issue_id
+        self.schema_version = schema_version
+        self.customization_args = customization_args
+
+    def to_dict(self):
+        """Returns a dict representation of the ExplorationIssue domain object.
+
+        Returns:
+            dict. A dict mapping of all fields of ExplorationIssue object.
+        """
+        return {
+            'issue_id': self.issue_id,
+            'schema_version': self.schema_version,
+            'customization_args': (
+                {} if self.issue_id is None else
```

Comment:
Why do we need this check here? Wouldn't `validate` fail if `issue_id` were `None`?

Reply:
@brianrodri . We're technically supposed to validate the domain class before using the method to_dict(), but we can't always bank on the client to do it perfectly right? I don't think its an issue having an additional check just in case.

Reply:
Hmm it only bothers me b/c we don't check anywhere else. I'll defer to @anmolshkl for a second opinion.

Reply:
@pranavsid98 I agree with @brianrodri here. `to_dict()` should do only one job i.e. converting the object to its dict representation. The validation should be done by the client itself (perhaps the service layer) in which case it will mostly be written by you. For example, take a look [here](https://github.com/oppia/oppia/blob/c8eae85697cb68c206042ebdaf74d1dd4f3aec80/core/domain/classifier_services.py#L227).

Reply:
Makes sense. Done.